store.go

```diff
@@ -252,6 +252,46 @@ func (store *Store) clean(key string) string {
 	return key
 }
 
+// Our own snowflake cleanPath, derived from net/http/server.go
+func cleanPath(p string) string {
+	if p == "" {
+		return ""
+	}
+
+	// Convert trailing /. to /, so path.Clean doesn't get to it
+	if strings.HasSuffix(p, "/.") {
+		p = p[:len(p)-1]
+	}
+
+	if p[0] != '/' {
+		p = "/" + p
+	}
+	np := path.Clean(p)
+	// path.Clean removes trailing slash except for root;
+	// put the trailing slash back if necessary.
+	if p[len(p)-1] == '/' && np != "/" {
+		np += "/"
+	}
+	return np
+}
+
+func formatKVPath(prefix, key string) string {
+	joined := cleanPath("/v1/kv/"+prefix) + "/" + cleanPath(key)
```

Comment:
Should this be `"/v1/kv/" + cleanPath(prefix)` ? So if `prefix` contains `..` it doesn't remove the `/v1/kv` root?

Reply:
Yes, good catch.